### ISSUE TITLE
Slow PDF OCR Processing

### ISSUE THREAD

**Issue by ffalkenberg** [h2oai/h2ogpt#811]
I am using the following command to start my server:

```bash
python3.10 /workspace/generate.py \
    --base_model=meta-llama/Llama-2-13b-chat-hf \
    --prompt_type=llama2 \
    --save_dir='/workspace/save/' \
    --score_model=None \
    --max_max_new_tokens=2048 \
    --max_new_tokens=1024 \
    --top_k_docs=3 \
    --pre_load_caption_model=True \
    --load_8bit=True \
    --enable_ocr=True \
    --enable_captions=True \
    --enable_pdf_ocr=True \
    --enable_pdf_doctr=True \
    --enable_doctr=True \
    --enable_pix2struct=True \
    --max_quality=True
```

#### Problem:

When I upload a multipage PDF for OCR processing, it takes up to 55 seconds to complete the OCR. This seems quite slow and affects the user experience. As a reference, I am using [this sample PDF](https://github.com/fraponyo94/Text-Extraction-Scanned-Pdf/blob/master/sample-scanned-pdfs/PublicWaterMassMailing.pdf).

#### Questions:

1. Is this upload and processing time plausible given the configuration I am using?
2. Are there any ways to optimize the process for faster upload and OCR processing?
3. Can the processing for multiple pages be done in parallel?

#### Environment:

- GPU: NVIDIA A100
- CPU: Intel Xeon Platinum 8362 CPU with 8 cores

Thank you for your time and assistance.

**Comment by ffalkenberg** [comment]
@pseudotensor 
Thank you for the feedback, i am looking into this.
As a first step I've created a pull request for a related bugfix [Pull Request #819](https://github.com/h2oai/h2ogpt/pull/819)

**Comment by pseudotensor** [comment]
Already done for doctr pdf case
